### ISSUE TITLE
just sharing some Perl code (failed try to recreate a schematic)...

### ISSUE THREAD

**Issue by kittennbfive** [misterblack1/plexus-p20#16]
Hi Adrian, thanks for you fascinating videos, especially the last one (missing in the Readme, i created a PR)!

I thought i share some code (Perl) i wrote as i tried to (ab)use Graphviz to turn the stuff you found back into a (kindof) schematic. The result is a big "ball of wool" and basically useless except maybe as modern art, but maybe somebody can do *something* with this stuff/code. No copyright claimed as long as your doing peaceful stuff with it.

I also used this code to automatically "guess" the number of pins of each part. For this i assumed that the *last* pin of $component is connected *somewhere*, so i iterated over all nets and updated the pincount of $component if the new value is bigger than the existing. Pretty basic but doing this by hand would take some time, so maybe it is of some use.

Files attached. Be warned it was a quick and dirty try, the code works but isn't pretty...

![mess](https://github.com/misterblack1/plexus-p20/assets/58293822/9d2fd8a2-f989-4718-80ab-d10f466409ec)
[number_of_pins.txt](https://github.com/user-attachments/files/15904716/number_of_pins.txt)
[graph.pl.txt](https://github.com/user-attachments/files/15904719/graph.pl.txt)

**Comment by Hannibal-pl** [comment]
I suspect that in this cad file has nothing in common with schematics. It probably only crude version of current boardviews. Electronic parts are placed in ridiculous way.

I also try to draw this as svg file, but I stop at putting rectagles of shape of IC's on canvas. It look nor as schematics nor as boardview. It seems to be randomly placed in more-less horizontal lines. 

You can see it in my repo: https://github.com/Hannibal-pl/plexus_cad2svg - I have no plans to develop it further, as it has no sense.

**Comment by kittennbfive** [comment]
I am sorry, i did not understand. :(

The netlist is a representation of all the electrical connections like the schematic; however in the schematic busses are drawn as a single (big) line (or something similar) and in the netlist everything is split apart / all the *individual* connections are listed. I suspect that this is at least one reason the output of my tool is completly unusable.

It should be possible to detect busses and draw them in a better way; however as the board has been fixed by Adrian (good job btw) i won't invest more time in writing code for this.

**Comment by Hannibal-pl** [comment]
Yes, you can draw connections in better, but this give nothing when parts are placed in somewhat random way. Putting most IC in 5, 6 rows with tens of it in each or grouping all capacitors together is not how you draw schematics. Clearly this is not boarviedw or PCB design also. IMHO this was only this for what Adrian is use this data - fast way to find all connections form selected pin. Original authors probably abuse some CAD software to do this, so they not care where they place parts. Only important thing for them was connections.

**Comment by Qyon** [comment]
I'v tried to convert it to kicad schematic. Maybe someone is interested. https://github.com/Qyon/CalayCADConverterResults
![Image](https://github.com/user-attachments/assets/fd8362e1-6ce6-4804-bc33-d0357eb80788)
[out2.kicad_sch.pdf](https://github.com/user-attachments/files/18983896/out2.kicad_sch.pdf)
